argcomplete/__init__.py

```diff
@@ -321,18 +332,10 @@ def __call__(self, parser, namespace, values, option_string=None):
         return self.active_parsers
 
     def _get_subparser_completions(self, parser, cword_prefix):
-        def filter_aliases(metavar, dest, prefix):
-            if not metavar:
-                return dest if dest and dest.startswith(prefix) else ""
-
-            # metavar combines dest and aliases with ",".
-            a = metavar.replace(",", "").split()
-            return " ".join(x for x in a if x.startswith(prefix))
```

Comment:
This was handling subparser aliases. Example from https://docs.python.org/3/library/argparse.html:

```
>>> parser = argparse.ArgumentParser()
>>> subparsers = parser.add_subparsers()
>>> checkout = subparsers.add_parser('checkout', aliases=['co'])
>>> checkout.add_argument('foo')
>>> parser.parse_args(['co', 'bar'])
Namespace(foo='bar')
```

It stores the display for these in the metavar attribute as e.g. "checkout (co)". If there's more than one, it joins them with a comma, e.g. "checkout (co, other)".

The removed code doesn't actually handle this correctly because it doesn't account for the braces. A more robust solution might be to inspect `parser._name_parser_map` to see which names map to the same parser as `parser._name_parser_map[dest]`. If we want to maintain strict backwards compatibility, we'd have to also store the existing display text alongside the tuple, otherwise we can't guarantee we'd reconstruct it in the correct order. Alternatively if we're happy to accept a minor break, you could just join the names back together as you are with the others.

Reply:
Thank you for explanation, covered current behaviour in https://github.com/kislyuk/argcomplete/pull/295/files

Reply:
Added ugly fix `a = metavar.replace(" (", ", ").replace(")", "").split(", ")` which guarantees order but still break compatibility because of missing braces.

Reply:
I rewritten code using `parser.choices`. @evanunderscore take a look, please

Reply:
Looks good to me, thanks!

Reply:
I've merged your other PR with the test case. Can you merge/rebase this branch so that the change is more obvious in the diff please?

Reply:
done